src/WPF/WPF.Viewer/Samples/NetworkAnalysis/FindServiceArea/FindServiceArea.xaml.cs

```diff
@@ -50,46 +53,22 @@ private void Initialize()
 
             // Create graphics overlays for all of the elements of the map.
             MyMapView.GraphicsOverlays.Add(new GraphicsOverlay());
-
-            // Add a new behavior for double taps on the MapView.
-            MyMapView.GeoViewDoubleTapped += (s, e) =>
-            {
-                // If the sketch editor complete command is enabled, a sketch is in progress.
-                if (MyMapView.SketchEditor.CompleteCommand.CanExecute(null))
-                {
-                    // Set the event as handled.
-                    e.Handled = true;
-
-                    // Finish the drawing.
-                    MyMapView.SketchEditor.CompleteCommand.Execute(null);
-                    DrawBarrierButton.Content = "Draw barrier";
-                }
-            };
         }
 
-        private async void PlaceFacilityButton_Click(object sender, RoutedEventArgs e)
+        private void PlaceFacilityButton_Click(object sender, RoutedEventArgs e)
         {
             try
             {
-                // Let the user tap on the map view using the point sketch mode.
-                SketchCreationMode creationMode = SketchCreationMode.Point;
-                Geometry geometry = await MyMapView.SketchEditor.StartAsync(creationMode, false);
-
-                // Symbology for a facility.
-                PictureMarkerSymbol facilitySymbol = new PictureMarkerSymbol(new Uri("https://static.arcgis.com/images/Symbols/SafetyHealth/Hospital.png"))
+                if (MyMapView.GeometryEditor.IsStarted)
```

Comment:
You can ignore this feedback. Your changes look consistent in that you handle completed geometry in `GeometryEditor.PropertyChanged` handler so the Stop() here should trigger that event.

This suggestion was in case you wanted to keep the following syntax:
```CSharp
var geometry  = await StartAsync(..)
// do something with geometry
```
to do that, you can use `TaskCompletionSource<Geometry> _tcs` that you set result in the event handler so it becomes
```CSharp
var geometry = await _tcs.Task; // will return once TrySetResult/Exception/Canceled are called.
```

I think how you translated legacy code is just as valid, clean and consistent. Marking this as resolved.